serf/serf.go

```diff
@@ -593,9 +595,9 @@ func (s *Serf) Join(existing []string, ignoreOld bool) (int, error) {
 	// Ignore any events from a potential join. This is safe since we hold
 	// the joinLock and nobody else can be doing a Join
 	if ignoreOld {
-		s.eventJoinIgnore = true
+		s.eventJoinIgnore.Store(true)
```

Comment:
It wasn't clear to me why this needed another locking mechanism when its already in a block that's guarded by the join lock in line # 592